### PR TITLE
[ADP-3110] Move primitive types to the primitive library

### PR DESCRIPTION
**Pull request by paolino**
Within the partial goal of separating the Address hierarchy in a sub-library, these types seemed  to belong to the primitive lib where they can be abandoned in favor of ledger types one day. In the meantime Address hierarchy can use them breaking the circular dependency with the "primitive" in the _cardano-wallet_ that depends on the incoming _cardano-wallet-address_ sub-library

- [x] move NetworkId
- [x] move RewardAccount
- [x] move ProtocolMagic
- [x] move Tx

ADP-3110

### CODE REVIEW

File: lib/wallet/src/Cardano/Wallet/Primitive/TxSeq/Gen.hs
Comment (HeinrichApfelmus) [inline lib/wallet/src/Cardano/Wallet/Primitive/TxSeq/Gen.hs:1]: I cannot find the corresponding deletion in `lib/wallet`.

File: lib/wallet/src/Cardano/Wallet/Primitive/TxSeq.hs
Comment (HeinrichApfelmus) [inline lib/wallet/src/Cardano/Wallet/Primitive/TxSeq.hs:1]: I cannot find the corresponding deletion in `lib/wallet`.

File: lib/primitive/lib/Cardano/Wallet/Primitive/Types/Tx/Metadata/Gen.hs
Comment (HeinrichApfelmus) [inline lib/primitive/lib/Cardano/Wallet/Primitive/Types/Tx/Metadata/Gen.hs:1]: I cannot find the corresponding deletion in `lib/wallet`. 🤔

**Review by HeinrichApfelmus — COMMENTED** [review]
Thanks! 😊 On the 100 files changed, I can't find the deletions corresponding to 3 files of the `.Gen` variety, though. 🤔

File: lib/wallet/test/unit/Cardano/Wallet/DB/StateMachine.hs [hunk 1]
```diff
@@ -136,28 +136,22 @@ import Cardano.Wallet.Primitive.Types.TokenPolicy
...
-    , TransactionInfo (..)
-    , Tx (..)
-    , TxCBOR
-    , TxMeta
-    , TxMetadata
-    , TxScriptValidity
-    , TxStatus
-    , inputs
-    )
 import Cardano.Wallet.Primitive.Types.Tx.Constraints
```
Comment (paolino) [inline lib/wallet/test/unit/Cardano/Wallet/DB/StateMachine.hs:147]: Explicit imports missing

File: lib/wallet/integration/src/Test/Integration/Scenario/API/Shelley/TransactionsNew.hs [hunk 2]
```diff
@@ -235,6 +233,8 @@ import qualified Cardano.Ledger.Keys as Ledger
 import qualified Cardano.Wallet.Address.Derivation.Shelley as Shelley
 import qualified Cardano.Wallet.Api.Link as Link
 import qualified Cardano.Wallet.Primitive.Types.TokenMap as TokenMap
+import Cardano.Wallet.Primitive.Types.Tx.TxMeta
+    ( Direction (..), TxStatus (..) )
 import qualified Data.Aeson as Aeson
```
Comment (paolino) [inline lib/wallet/integration/src/Test/Integration/Scenario/API/Shelley/TransactionsNew.hs]: Misplaced import

**Review by HeinrichApfelmus — APPROVED** [review]
Looks good to me now. 😊

File: lib/wallet/integration/src/Test/Integration/Scenario/API/Shelley/TransactionsNew.hs [hunk 3]
```diff
@@ -1430,7 +1430,7 @@ spec = describe "NEW_SHELLEY_TRANSACTIONS" $ do
 
         -- transaction setting up reference script
         -- Build:
-        -- $ cardano-cli transaction build \
+        -- \$ cardano-cli transaction build \
```
Comment (jonathanknowles) [inline lib/wallet/integration/src/Test/Integration/Scenario/API/Shelley/TransactionsNew.hs]: Hi @paolino 

Is this change intentional?

There are also several other similar changes in this file.

File: nix/project-package-list.nix [hunk 4]
```diff
@@ -1 +1 @@
-[ "cardano-balance-tx" "cardano-coin-selection" "cardano-numeric" "cardano-wallet" "cardano-wallet-launcher" "cardano-wallet-primitive" "cardano-wallet-test-utils" "delta-store" "delta-table" "delta-types" "text-class" "wai-middleware-logging" ]
+[ "cardano-balance-tx" "cardano-coin-selection" "cardano-numeric" "cardano-wallet" "cardano-wallet-benchmarks" "cardano-wallet-launcher" "cardano-wallet-primitive" "cardano-wallet-test-utils" "delta-store" "delta-table" "delta-types" "text-class" "wai-middleware-logging" ]
```
Comment (jonathanknowles) [inline nix/project-package-list.nix:1]: Hi @paolino 

Is this addition of `cardano-wallet-benchmarks` intentional?

Asking as it seems to be unrelated to the PR title and description.

**Review by jonathanknowles — COMMENTED** [review]
Hi @paolino 

This looks fine to me. However, there seem to be some unintentional changes in [lib/wallet/integration/src/Test/Integration/Scenario/API/Shelley/TransactionsNew.hs](https://github.com/cardano-foundation/cardano-wallet/pull/4079/files/5aaf9c6dbedebcfc805a42e82ef09b00b7226e55#diff-dab339352a8b7603781e358f5150ba2f789870c8fef32f130b2efb561dffa595). Are these changes intentional? (See comments.)

File: lib/wallet/integration/src/Test/Integration/Scenario/API/Shelley/TransactionsNew.hs (hunk 3 above)
Comment (paolino) [reply lib/wallet/integration/src/Test/Integration/Scenario/API/Shelley/TransactionsNew.hs]: oh, that's really unrelated. It fixed some warnings, but I can roll them back.

File: nix/project-package-list.nix (hunk 4 above)
Comment (paolino) [reply nix/project-package-list.nix:1]: mhh, I think there was a request from @HeinrichApfelmus to run nix/regenerate.sh , or maybe I confused it from the previous PR. I didn't edit this by hand if this is the question.

File: lib/wallet/integration/src/Test/Integration/Scenario/API/Shelley/TransactionsNew.hs (hunk 3 above)
Comment (paolino) [reply lib/wallet/integration/src/Test/Integration/Scenario/API/Shelley/TransactionsNew.hs]: removed

File: nix/project-package-list.nix (hunk 4 above)
Comment (paolino) [reply nix/project-package-list.nix:1]: Moved to a new commit. On a  tangent note, CI doesn't require we run nix/regenerate.sh but I think it should as it's doing it before running its steps.

Comment (HeinrichApfelmus) [reply nix/project-package-list.nix:1]: > Is this addition of cardano-wallet-benchmarks intentional?
> Asking as it seems to be unrelated to the PR title and description.

It's unrelated, but should be done — perhaps in a different PR.

I think that `cardano-wallet-read` should also be added.

The issue is that `nix/project-package-list.nix` influences which tests are run. I don't quite understand the inner workings of `./nix/regenerate.sh`, and I'm not sure why `cardano-wallet-read` was not added. 🤔

**Review by jonathanknowles — APPROVED** [review]
